### PR TITLE
change(ios): workspace reorganization + cleanup

### LINKED ISSUE #4613: chore(ios): workspace reorganization
**Is your feature request related to a problem? Please describe.**
The Xcode workspace for the iOS engine/app could _really_ use some reorganization.  I'm constantly having to figure out exactly where specific source files can be found within it... and I'm its main developer.  That organization _also_ fails to directly match the actual file system layout, which sometimes causes its own issues.

Doesn't help that Xcode doesn't auto-alphabetize things, either.

### PR DESCRIPTION
**Pull request by jahorton**
Fixes #4613.

In the process of reorganizing things, I found a few files that were dead code and thus safe to delete.  Aside from that, this reorganizes the source considerably, based on the following goals:

1. Where possible, the organization patterns of KeymanEngine and Keyman should match.
2. Entries within each "group" / folder should be alphabetized.
3. Files should be grouped based on purpose, not unlike with namespaces... even if Swift doesn't do namespaces.
4. "Groups" / folders before leaf files

### CODE REVIEW

**Comment by darcywong00** [comment]
InfoViewController.swift has a merge conflict

**Comment by jahorton** [comment]
> InfoViewController.swift has a merge conflict

Yeah, that's kind of expected with all the iOS branches I've put up lately.  I feel this PR to be lower priority than the other iOS PRs, so I'll wait to update this one until the others are merged.

File: ios/engine/KMEI/KeymanEngine/Classes/Installation/AssociatingPackageInstaller.swift [hunk 1]
```diff
@@ -7,6 +7,7 @@
 //
 
 import Foundation
+import UIKit
```
Comment (mcdurdin) [inline ios/engine/KMEI/KeymanEngine/Classes/Installation/AssociatingPackageInstaller.swift:10]: I'm puzzled by this change

**Review by mcdurdin — APPROVED** [review]
Good I think, though I hate to think what it'll be like cherry-picking changes to stable-14.0 if we have to do that 😁

File: ios/engine/KMEI/KeymanEngine/KeymanEngine.h [hunk 2]
```diff
@@ -1,17 +0,0 @@
-//
-//  KeymanEngine.h
-//  KeymanEngine
-//
-//  Created by Gabriel Wong on 2017-10-02.
-//  Copyright © 2017 SIL International. All rights reserved.
-//
-
-#import <UIKit/UIKit.h>
```
Comment (jahorton) [inline ios/engine/KMEI/KeymanEngine/KeymanEngine.h:9]: This line had (effectively) automatically added an `import UIKit` for all KeymanEngine source files.  With it gone, we need to be explicit in all cases where we reference `UIKit`.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.58-alpha](https://keyman.com/downloads/releases/alpha/15.0.58)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.59-alpha](https://keyman.com/downloads/releases/alpha/15.0.59)

**Pull request merged**